include/vkgcDefs.h

```diff
@@ -811,7 +812,7 @@ struct PipelineShaderOptions {
   unsigned ldsSpillLimitDwords;
 
   /// Attempt to scalarize waterfall descriptor loads.
-  bool scalarizeWaterfallLoads;
+  std::optional<bool> scalarizeWaterfallLoads;
```

Comment:
I did this because we decided that Vkgc::PipelineShaderOptions::scalarizeWaterfallLoads really needs to be tri-state:

Use compiler default
Force scalarization on
Force scalarization off

Reply:
@nhaehnle : Should I keep it as std::optional<bool> or change to enum?

Reply:
I'd rather use an enum, and we could write some macro to auto-generate these enums if this is a common pattern.